server/src/main/java/io/crate/planner/operators/MultiPhase.java

```diff
@@ -57,9 +56,7 @@ public static LogicalPlan createIfNeeded(Map<LogicalPlan, SelectSymbol> uncorrel
 
     private MultiPhase(LogicalPlan source, Map<LogicalPlan, SelectSymbol> subQueries) {
         super(source);
-        HashMap<LogicalPlan, SelectSymbol> allSubQueries = new HashMap<>(source.dependencies());
-        allSubQueries.putAll(subQueries);
-        this.subQueries = Collections.unmodifiableMap(allSubQueries);
+        this.subQueries = subQueries;
```

Comment:
subQueries is only used for `dependencies()` and should be calculated based on the given source.